frontend/src/lib/k8s/networkpolicy.tsx

```diff
@@ -0,0 +1,41 @@
+import { apiFactoryWithNamespace } from './apiProxy';
+import { KubeObjectInterface, LabelSelector, makeKubeObject } from './cluster';
+
+interface NetworkPolicyPort {
+  port: string | number;
+  protocol: string;
```

Comment:
I think this is missing `endPort`?
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#networkpolicyport-v1-networking-k8s-io

All fields also seem to be optional (depending on their combination) in the spec.